x/protocolpool/types/params.go

```diff
@@ -0,0 +1,29 @@
+package types
+
+import (
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func DefaultParams() Params {
+	return Params{
+		EnabledDistributionDenoms: []string{sdk.DefaultBondDenom},
+		DistributionFrequency:     1,
+	}
+}
+
+func (params *Params) Validate() error {
+	for _, d := range params.EnabledDistributionDenoms {
+		err := sdk.ValidateDenom(d)
+		if err != nil {
+			return err
+		}
+	}
+
+	if params.DistributionFrequency == 0 {
+		return errors.New("DistributionFrequency must be greater than 0")
+	}
+
+	return nil
+}
```

Comment:
check for duplicate denom?

Reply:
https://github.com/cosmos/cosmos-sdk/pull/24116/commits/a0c82e6e223695e5e694941a352dcb74705b4342